src/main/java/fruitshop/strategy/ReturnOperation.java

```diff
@@ -0,0 +1,20 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class ReturnOperation implements OperationHandler {
+    private final Storage storage = new Storage();
+
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        int newQuantity = fruitTransaction.getQuantity()
+                + storage.getQuantity(fruitTransaction.getFruit());
+        if (newQuantity >= 0) {
+            storage.put(fruitTransaction.getFruit(), newQuantity);
+        } else {
+            throw new RuntimeException("negative balance cannot be recorded at "
+                    + ReturnOperation.class);
+        }
+    }
+}
```

Comment:
The `ReturnOperation` class should not instantiate a new `Storage` object. Instead, it should receive a reference to an existing `Storage` object, for example, through its constructor. This will ensure that all operations are performed on the same storage and adhere to the SOLID principles, particularly the Dependency Inversion Principle.